### DISCUSSION TITLE
Impossible to Login on LTSP Client : read:Connection reset by peer. Login incorrect.

### DISCUSSION THREAD

**Discussion by Patrick93600** [ltsp/ltsp#333] [Q&A]
Hello to All,
I'm very new to LTSP and try to install a fresh configuration as test bed for now. 
I've followed the full installation, still i'm unable to login on the LTSP Client.

Configuration is as such : 
LTSP Server : is an Ubuntu Mate 20.04 VirtualBox VM running from a Ubuntu 20.04 Server. VirtualBox is latest 6.16
LTSP Client : a netboot PXE client, generated from "ltsp image /". Client is booting fine, till the splash screen where i'm unable to login. I've tried the console mode (Ctrl-Alt-F2) and received exactly the same answer "read:Connection reset by peer. Login incorrect.". 

I must have done something wrong, miss a step, as it looks nobody else is encountering same issue. (search the discussions with no luck). 
If somebody can direct me to a solution for this issue.
Thanks for your kind help in advance.
Regards,
Patrick.

**Comment by alkisg** [comment]
This usually means that either the client doesn't have the SSHFS package, or the user is incorrect. The user may be incorrect if LTSP administrators haven't run `ltsp initrd`, or when they're using a chroot/image user, instead of a server user.

For troubleshooting, the last line of https://ltsp.org/man/ltsp.conf/ says how to login to the client as root. Do that, and then try to `ssh user@server`, or to `login user`, and then check the various logs.

**Reply by Patrick93600** [reply]
Hello Alksig,
Thanks for the speedy response yesterday.
Actually, i guess i found the root cause of my problem.. which indeed is close to sshd_config configuration.
I'm normally using a different port than tcp/22 for ssh to my servers... this was the case for my LTSP server as well.
While redoing everything from scratch, i did kept the standard configuration on sshd, (which by default, when all is commented, activate on port tcp/22)... and it worked like a charm. 
I've done several tests, including having sshd_config with several ports at the same time (include port 22, and others) to try to debug :  ssh to server is working fine on each ports,  however, next connection to client session is failing afterwards. (even after refreshing image, initrd, ..etc). 
I was not able to configure LTSP to make it working properly with different port than tcp/22 ( if i'm removing all extract ports on sshd_config, it works again).
May be there are some further parameters to allow... 
Thanks for your help.
Regards
Patrick.

**Comment by alkisg** [comment]
To change the SSH port, read https://github.com/ltsp/ltsp/issues/66.

**Reply by Patrick93600** [reply]
Hello 
I've created a file base on the Wiki elements :  install -m 0660 -g sudo /usr/share/ltsp/common/ltsp/ltsp.conf /etc/ltsp/ltsp.conf
then added "POST_INIT_SSHD_PORT="sed 's/NumberOfPasswordPrompts=1/& -p 22000/' -i /usr/share/ltsp/client/login/pamltsp""on the end of the file. (where 22000 is corresponding to sshd_config port directive)
Recreate image, update ltsp ipxe, nfs, initrd... 
However, this goes back to the previous error, mention as title for this discussion.
Is there something else i'm missing ?
Thanks 
Patrick.

**Comment by alkisg** [comment]
Do not put it at the end of the ltsp.conf file, put it in the [clients] section.
You can log in as root on the client and run `grep 22000 /usr/share/ltsp/client/login/pamltsp` to see if you actually managed to update that file.

**Reply by Patrick93600** [reply]
Hello,
I've move the above line to the [clients] section and i now works as planned with the sshd_config with port 22000.
However the /usr/share/ltsp/client/login/pamltsp does not show any line containing 22000. (despite it works).

As per a try, i would like to forbid the usage of clear text passwords on the server and replace by a key challenge. As i'm using the standard chrootless method to generate the image... all i've done in the server is replicated to the clients :  I've tried to restate PasswordAuthentication to "no" into sshd_config, but then back to impossible to login to the clients.
Any clue ?
Thanks.
Patrick

**Comment by alkisg** [comment]
Key challenge means that the user is ALREADY logged in, has a /home/username directory mounted, and inside it, there's a .ssh/id_rsa file with the key.

In your case, how would /home/username get mounted before the user enters the password?

When LTSP clients boot, they can have no passwords and no keys in the disk, because it's a public image served by NFS. The first "secret" that they get is the user password.

You could use USB sticks for authentication though...
